### ISSUE TITLE
Known built-in script limitations

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#55287]
### Godot version

efd0db8

### System information

W10

### Issue description

Here's a list of known limitations of built-in scripts. It might not be complete, because built-in scripts have more issues, but the others I found seem fixable. If they turn out otherwise, I will add them here later.

We should make sure that the users know these limitations. My idea was to add a big tooltip/dialog in the script creation dialog when you click something, e.g. a button next to this warning:
![image](https://user-images.githubusercontent.com/2223172/143243377-fd034f13-f423-43f3-9fac-403999094e56.png)

The list is (including explanations and theoretical fixes):

1. _Can't open built-in scripts in external editor._ They are saved inside scene, so to edit them, you need to edit scene file.

It *is* possible to extract built-in script into temporary file and synchronize with the scene, but it's ultra hacky and overkill to implement.

2. _Can't use `class_name` in bult-in scripts._ Class names are registered as references to script files, so in this case the class would need to point to a resource inside scene. Even if it was possible, you'd need to load the whole scene each time you want to parse it.
3. _Can't create script documentation for built-in scripts._ Same reason as above. Docs are created based on script files.
4. _Built-in scripts don't list in scene dependencies if they preload something._ Again, same as above. The ResourceLoader will parse every external script referenced in the scene looking for preloads. But built-in scripts require parsing the scene itself.
5. _Built-in scripts can't be extended nor preloaded._ You can't do `extends Scene.tscn::scriptid`. Same reason as above. Parser doesn't recognize internal paths. Interestingly, using `load()` will work.

2, 3, 4 and 5 could be fixed if we had a way to partially parse a scene to load a built-in resource. For text scenes it might not be a problem, but binary scenes could be more problematic, idk. Also internal resource paths need to be recognized by the parser (right now it assumes actual file paths for `extend`, `preload` etc).
But even then, fixing 2 and 3 could cause performance issues, because every scene would need to be parsed in search of built-in scripts, so it's likely not feasible.

### Steps to reproduce

1. Create built-in script.

**Comment by KoBeWi** [comment]
Right, I should add it too. C# for example needs a script file to compile.

**Comment by Listwon** [comment]
`Search>Find In Files...` inside Godot doesn't show results from built-in scripts.

**Comment by KoBeWi** [comment]
> Search>Find In Files... inside Godot doesn't show results from built-in scripts.

This is not a limitation. Built-in scripts are inside scenes, so you need to add tscn to search extensions in Project Settings.

**Comment by Demozo** [comment]
https://github.com/godotengine/godot/issues/64213 also caused by a limitation of built-in scripts.

**Comment by KoBeWi** [comment]
#64213 is a bug (and a duplicate).

**Comment by HolonProduction** [comment]
> 5\. using `load()` will work.

Depends see https://github.com/godotengine/godot/issues/116593
